presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -679,6 +723,12 @@ public String toString()
         }
     }
 
+    private static enum NullPartitioningPolicy
+    {
+        NONE,
```

Comment:
I'd name these something like `ONE_PLACE` and `EVERYWHERE`, but will defer to @martint for naming the various properties